DiceKit/SubtractionExpression.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  SubtractionExpression.swift
+//  DiceKit
+//
+//  Created by Jonathan Hoffman on 8/15/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import Foundation
+
+public struct SubtractionExpression<LeftExpression: protocol<ExpressionType, Equatable>, RightExpression: protocol<ExpressionType, Equatable>>: Equatable {
+    
+    public let minuend: LeftExpression
+    public let subtrahend: RightExpression
+    
+    public init(_ minuend: LeftExpression, _ subtrahend: RightExpression) {
+        self.minuend = minuend
+        self.subtrahend = subtrahend
+    }
+    
+}
+
+// MARK: - ExpressionType
+
+extension SubtractionExpression: ExpressionType {
+    
+    public typealias Result = SubtractionExpressionResult<LeftExpression.Result, RightExpression.Result>
+    
+    public func evaluate() -> Result {
+        let leftResult = minuend.evaluate()
+        let rightResult = subtrahend.evaluate()
+        
+        return Result(leftResult, rightResult)
+    }
+    
+    public var probabilityMass: ExpressionProbabilityMass {
+        return minuend.probabilityMass && NegationExpression(subtrahend).probabilityMass
+    }
+    
+}
+
+// MARK: - CustomStringConvertible
+
+extension SubtractionExpression: CustomStringConvertible {
+    
+    public var description: String {
+        return "\(minuend) - \(subtrahend)"
+    }
+    
+}
+
+// MARK: - CustomDebugStringConvertible
+
+extension SubtractionExpression: CustomDebugStringConvertible {
+    
+    public var debugDescription: String {
+        return "\(String(reflecting: minuend)) - \(String(reflecting: subtrahend))"
+    }
+    
+}
+
+// MARK: - Equatable
+
+private func equate<L, R>(lhs: SubtractionExpression<L, R>, _ rhs: SubtractionExpression<L, R>) -> Bool {
```

Comment:
I was following the pattern in `AdditionExpression`. I can change `SubtractionExpression` to not use this function, then change addition in a new PR, maybe the same one where I add commutativity to the addition result type.